src/strings/ops.c

```diff
@@ -644,7 +642,34 @@ MVMint64 MVM_string_equal_at_ignore_case(MVMThreadContext *tc, MVMString *haysta
     }
     return 1;
 }
+MVMint64 MVM_string_index_ignore_case(MVMThreadContext *tc, MVMString *haystack, MVMString *needle, MVMint64 start) {
+    size_t index           = (size_t)start;
+    MVMStringIndex hgraphs = MVM_string_graphs(tc, haystack), ngraphs = MVM_string_graphs(tc, needle);
+
+    MVM_string_check_arg(tc, haystack, "index search target");
+    MVM_string_check_arg(tc, needle, "index search term");
+
+    if (!ngraphs)
+        return start <= hgraphs ? start : -1; /* the empty string is in any other string */
+
+    if (!hgraphs)
+        return -1;
+
+    if (start < 0 || start >= hgraphs)
+        return -1;
+
+    if (ngraphs > hgraphs || ngraphs < 1)
+        return -1;
 
+    /* brute force for now. horrible, yes. halp. */
+    while (index <= hgraphs - ngraphs) {
+        if (MVM_string_equal_at_ignore_case(tc, haystack, needle, index)) {
```

Comment:
I believe `MVM_string_equal_at_ignore_case` allocates due to doing an `fc` operation on `needle`. Thus `MVMROOT` of both haystack and needed are required. The most efficient way is to place them around the whole loop, however the loop should then not `return`, but rather set a result and break, otherwise we'd not clear the rooted things from the temporary stack.